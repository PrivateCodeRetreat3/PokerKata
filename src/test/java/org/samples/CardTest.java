package org.samples;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class CardTest {
    @Test
    void testCardDeck() {
        Approvals.verifyAll("card", Deck.all() );
    }

    @Test
    void testWinners() {
        Hand h1 = new Hand("A♦");
        Hand h2 = new Hand("K♦");
        Hand[] hands = {h1, h2};
        CombinationApprovals.verifyAllCombinations((h11, h21) -> {
            if(h11 == h21)
            {
                throw new org.approvaltests.combinations.SkipCombination();
            }
            return Poker.whoWins(h11, h21);
        }, hands, hands);

    }

    @Test
    void testHighCardWins() {

        CombinationApprovals.verifyAllCombinations((c1, c2) -> {
            if(c1==c2)
            {
                throw new org.approvaltests.combinations.SkipCombination();
            }
            return Poker.whoWins(new Hand(c1),new Hand(c2));
        }, Deck.all(), Deck.all());

    }
}
