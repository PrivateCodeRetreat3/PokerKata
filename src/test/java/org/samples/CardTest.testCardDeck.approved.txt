card[00] = 2♣
card[01] = 2♦
card[02] = 2♥
card[03] = 2♦
card[04] = 3♣
card[05] = 3♦
card[06] = 3♥
card[07] = 3♦
card[08] = 4♣
card[09] = 4♦
card[10] = 4♥
card[11] = 4♦
card[12] = 5♣
card[13] = 5♦
card[14] = 5♥
card[15] = 5♦
card[16] = 6♣
card[17] = 6♦
card[18] = 6♥
card[19] = 6♦
card[20] = 7♣
card[21] = 7♦
card[22] = 7♥
card[23] = 7♦
card[24] = 8♣
card[25] = 8♦
card[26] = 8♥
card[27] = 8♦
card[28] = 9♣
card[29] = 9♦
card[30] = 9♥
card[31] = 9♦
card[32] = T♣
card[33] = T♦
card[34] = T♥
card[35] = T♦
card[36] = J♣
card[37] = J♦
card[38] = J♥
card[39] = J♦
card[40] = Q♣
card[41] = Q♦
card[42] = Q♥
card[43] = Q♦
card[44] = K♣
card[45] = K♦
card[46] = K♥
card[47] = K♦
card[48] = A♣
card[49] = A♦
card[50] = A♥
card[51] = A♦
